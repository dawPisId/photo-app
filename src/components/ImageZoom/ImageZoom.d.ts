export interface ImageZoomProps {
  setZoomTrigger: (bool: boolean) => void;
  imgUrl: string;
  imgLabel: string;
}
