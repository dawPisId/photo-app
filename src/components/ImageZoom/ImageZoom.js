import { Background, CloseButton, Image, Label } from "./ImageZoom.styled";

import CloseIcon from "@material-ui/icons/Close";
import React from "react";

const ImageZoom = ({ setZoomTrigger, imgUrl, imgLabel }) => (
  <Background>
    <CloseButton
      onClick={(e) => {
        setZoomTrigger(false);
      }}
    >
      <CloseIcon />
    </CloseButton>
    <Image>
      <img src={imgUrl} />
    </Image>
    <Label>{imgLabel}</Label>
  </Background>
);

export default ImageZoom;
