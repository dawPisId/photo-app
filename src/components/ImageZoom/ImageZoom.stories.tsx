import { ComponentMeta, ComponentStory } from "@storybook/react";

import Grid from "../Grid/Grid";
import ImageZoom from "./ImageZoom";
import React from "react";

export default {
  title: "Photo App/ImageZoom",
  component: ImageZoom,
} as ComponentMeta<typeof ImageZoom>;

const Template: ComponentStory<typeof ImageZoom> = (args) => (
  <ImageZoom {...args} />
);

export const Main = Template.bind({});
Main.args = {
  setZoomTrigger: () => null,
  imgUrl: "https://via.placeholder.com/600/92c952",
  imgLabel: "one",
};
