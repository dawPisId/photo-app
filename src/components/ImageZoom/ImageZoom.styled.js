import styled from "styled-components";

export const CloseButton = styled.div`
  margin-top: 1rem;
  margin-right: 2rem;
  align-self: flex-end;
  svg {
    color: white;
    &.MuiSvgIcon-root {
      font-size: 2rem;
      display: block;
    }
  }
`;

export const Background = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  backdrop-filter: blur(5px);
  position: fixed;
  height: 100vh;
  width: 100vw;
  left: 0px;
  top: 0px;
  background-color: rgba(0, 0, 0, 0.7);
  z-index: 2;
`;

export const Image = styled.div`
  margin-top: 5rem;
  margin-bottom: 2rem;
  img {
    max-width: 20rem;
  }
  @media (max-width: 430px) {
    img {
      max-width: 17rem;
    }
  }
  @media (max-width: 330px) {
    img {
      max-width: 15rem;
    }
  }
`;

export const Label = styled.div`
  color:white;
  font-size: 1.25rem;
  font-family: "Roboto", sans-serif;
  @media(max-width:430px){
    font-size:1rem;
  @media(max-width:360px){
    font-size:0.8rem;
  }
`;
