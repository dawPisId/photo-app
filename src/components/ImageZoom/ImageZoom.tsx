import { Background, CloseButton, Image, Label } from "./ImageZoom.styled";
import React, { FC } from "react";

import CloseIcon from "@material-ui/icons/Close";
import { ImageZoomProps } from "./ImageZoom.d";

const ImageZoom: FC<ImageZoomProps> = ({
  setZoomTrigger,
  imgUrl,
  imgLabel,
}) => (
  <Background data-testid="testZoom">
    <CloseButton
      data-testid="testButton"
      onClick={(e) => setZoomTrigger(false)}
    >
      <CloseIcon />
    </CloseButton>
    <Image>
      <img src={imgUrl} alt="" />
    </Image>
    <Label>{imgLabel}</Label>
  </Background>
);

export default ImageZoom;
