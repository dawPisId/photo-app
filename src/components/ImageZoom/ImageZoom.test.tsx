import ImageZoom from "./ImageZoom";
import React from "react";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <ImageZoom
        setZoomTrigger={() => null}
        imgUrl="https://via.placeholder.com/600/92c952"
        imgLabel="one"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
