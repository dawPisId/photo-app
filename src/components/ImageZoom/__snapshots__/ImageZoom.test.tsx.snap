// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-gtsrHT iPBQld"
>
  <div
    className="sc-bdnxRM hnBNpD"
    onClick={[Function]}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
      />
    </svg>
  </div>
  <div
    className="sc-dlnjwi cLgzjL"
  >
    <img
      alt=""
      src="https://via.placeholder.com/600/92c952"
    />
  </div>
  <div
    className="sc-hKFxyN eUPgQk"
  >
    one
  </div>
</div>
`;
